mp4parse_capi/src/lib.rs

```diff
@@ -393,6 +393,15 @@ fn track_time_to_us(time: TrackScaledTime, scale: TrackTimeScale) -> Option<u64>
     rational_scale(time.0, scale.0, microseconds_per_second)
 }
 
+fn get_track_time_to_us(time: i64, scale: TrackTimeScale) -> i64 {
+    assert!(time >= 0);
+    let track_time = TrackScaledTime(time as u64, scale.1);
```

Comment:
This seems messy. I guess taking a specific type here preserves some type checking of the scale value, but it would be neater to keep a specific type for the track time itself. Can you make encapsulate the signed scaling in another type, perhaps using a `rational_scale_signed()` or making `rational_scale` generic? I'm not sure if there's a simple trait bound for integer types.